### PR TITLE
Repr and type hints added

### LINKED ISSUE #34: Add type hints
Add type hints to all classes and methods, using lowercase syntax.
Should include `from __future__ import annotations` at top of each file for backwards-compatibility to Python 3.7.

https://stackoverflow.com/questions/63939138/is-there-a-way-to-use-python-3-9-type-hinting-in-its-previous-versions

### LINKED ISSUE #37: Add __repr__ for all Pattern subclasses

### PR DESCRIPTION
**Pull request by loparcog**
To resolve #37,  `__repr__` function has been added to all Pattern classes and has been tested (with a handful of them) to confirm correct formatting. Testing consisted of making a Pattern object, pulling the representation, making a new object, and comparing the two:

```py
testPat = PExample(1, 2, 3)
newPat = eval(repr(testPat))
if (testPat == newPat):
    print("Success: %s" % repr(testPat))
```

I have also looked into adding type hints for any non-hidden functions, other than __init__, to resolve #34. For any numbers that didn't seem locked to integers I put float, and for any pattern inputs that didn't seem locked to patterns I put the base class of iter. 

There were some cases of key values and ambiguous types that I left empty since it seemed like a large range of types could be allowed. Feel free to peek in and let me know if there are any edits/adjustments to be made.

### CODE REVIEW

File: isobar/pattern/markov.py [hunk 1]
```diff
@@ -1,12 +1,14 @@
+from __future__ import annotations
 from .chance import PStochasticPattern
+from core import Pattern
```
Comment (ideoforms) [inline isobar/pattern/markov.py:3]: This should be .core

**Review by ideoforms — CHANGES_REQUESTED** [review]
Looking great and will definitely improve developer experience! 
A few things:

 - unit tests aren't passing at the moment because of the erroneous `from core` import in markov.py (comment inline)
 - type hints with a default value should always have a space around them, so `min:float=0.0, max:float=1.0` should become `min : float = 0.0, max : float = 1.0`. This could be automatically corrected across the codebase with `autopep8`, which is a really handy tool — or using command-option-L in PyCharm (which is my go-to quick autoformatter approach).
 -  I've spotted a couple of type hints that I'm not sure are correct: I believe `iter` should be  `typing.Iterable`, and `function` should be `typing.Callable`. But please do check those

**Comment by loparcog** [comment]
Got a new commit in addressing these:
- Fixed the import and unit tests are now working again
- Formatted all pattern files using `autopep8`, which seems to have matched your goal of argument formatting
- I was definitely wrong on the typing there, and have replaced the `iter` and `function` type hints with the proper `typing` types as suggested

File: isobar/pattern/fade.py [hunk 2]
```diff
@@ -107,10 +115,15 @@ class PFadeNotewiseRandom(PFadeNotewise):
     """ PFadeNotewiseRandom: Fade a pattern in/out by gradually introducing random notes. """
 
     def __init__(self, *args, **kwargs):
+        self.args = args
```
Comment (ideoforms) [inline isobar/pattern/fade.py:118]: Just spotted these new additions - what are they for?

File: isobar/pattern/sequence.py [hunk 3]
```diff
@@ -197,14 +214,17 @@ class PLoop(Pattern):
         [1, 4, 9, 1, 4, 9, 1, 4, 9, 1, 4, 9, 1, 4, 9, 1]
         """
 
-    def __init__(self, pattern, count=sys.maxsize):
+    def __init__(self, pattern: Iterable, count: int = sys.maxsize):
```
Comment (ideoforms) [inline isobar/pattern/sequence.py:217]: Just noticed that `pattern` is marked as an `Iterable` here and in a few other places later in the PR, but I think should always be a `Pattern`, right? (And is marked as a `Pattern` in most other places).

File: isobar/pattern/fade.py (hunk 2 above)
Comment (loparcog) [reply isobar/pattern/fade.py:118]: Saved the args so they could be used in the `repr` later on to easily reproduce the object. I'm realizing this just is a subclass of `PFadeNotewise` so would we wanna remove the `repr` function altogether for this class and have it default to the regular `PFadeNotewise` `repr`?

File: isobar/pattern/sequence.py (hunk 3 above)
Comment (loparcog) [reply isobar/pattern/sequence.py:217]: This was mainly used in places that looked like a list could work as well as a pattern input, I wasn't sure if we wanted to keep things more broad for this and meant to run this by you on if we should keep everything as `Pattern` in this instance

File: isobar/pattern/fade.py (hunk 2 above)
Comment (ideoforms) [reply isobar/pattern/fade.py:118]: ah I see! I think actually my implementation of `PFadeNotewiseRandom` was just a bit lazy, and should be corrected to be the same as `PFadeNotewise` :-)

Both of these PFade* classes are somewhat legacy and could do with revisiting someday...

File: isobar/pattern/sequence.py (hunk 3 above)
Comment (loparcog) [reply isobar/pattern/sequence.py]: Sounds good, this change has been reflected in the newest commit. Should any non-list params that seem to reference Patterns (like `input` for `PMap` as seen below, or `PScaleLinLin`) also be type hinted as a pattern?

```py
class PMap(Pattern):
    """ PMap: Apply an arbitrary function to an input pattern.
              Will pass any additional arguments, which can also be patterns.
              Instances of None in the input stream will be passed through to
              the function as normal.

        >>> PMap(PSeries(), lambda value: value * value).nextn(16)
        [0, 1, 4, 9, 16, 25, 36, 49, 64, 81, 100, 121, 144, 169, 196, 225]

        >>> PMap(PSeries(), pow, PSeries()).nextn(16)
        [1, 1, 4, 27, 256, 3125, 46656, 823543, 16777216, 387420489, 10000000000, ... ]
        """

    def __init__(self, input: Iterable, operator: Callable, *args, **kwargs):
```

File: isobar/pattern/fade.py (hunk 2 above)
Comment (loparcog) [reply isobar/pattern/fade.py]: I toyed around with `PFadeNotewiseRandom` to give it the proper params and map those properly to the `repr` function, let me know if you'd want me to adjust this back of if this is a decent solution for now

File: isobar/pattern/sequence.py (hunk 3 above)
Comment (ideoforms) [reply isobar/pattern/sequence.py]: Yep, sounds perfect!

Comment (loparcog) [reply isobar/pattern/sequence.py]: Implemented this in the newest commit

**Review by ideoforms — APPROVED** [review]
Finally merging this fine piece of work. Thank you!

**Pull request merged**